ports/nrf/common-hal/analogio/AnalogIn.c

```diff
@@ -76,7 +76,13 @@ uint16_t common_hal_analogio_analogin_get_value(analogio_analogin_obj_t *self) {
     // Something else might have used the ADC in a different way,
     // so we completely re-initialize it.
 
-    nrf_saadc_value_t value = -1;
+    // Adding the "asm volatile" memory fence here or many places after this declaration
+    // fixes an issue with gcc13 which causes `value` to always be zero.
+    // Compiling with gcc10 or gcc12 is fine.
+    // It can also be fixed by declaring `value` to be static.
+    // I think I'd like to declare this as volatile, but that causes type errors.
+    nrf_saadc_value_t value = 0;
+    asm volatile ("" : : : "memory");
```

Comment:
Should this go after the DMA (I assume) runs? Before we read it.

Reply:
That makes more sense, but having the `asm volatile` _anywhere_ after the declaration worked. So I moved it. But that may be an idiosyncrasy of the optimized code. By rights it should be declared volatile, but the `nrf` routines don't expect that declaration.

Reply:
Moved and re-tested to confirm.